### DISCUSSION TITLE
Do I need a pullSecret to install OKD

### DISCUSSION THREAD

**Discussion by roloff-it** [okd-project/okd#1930] [Q&A]
Hello

Until now I had understood that OKD was a free-to-install community project. Also free in the sense that I don't have to pay any license fees for an upstream project, like with CentOs Steam. After I looked at the install-config.yaml for OKD and also asked for a pullSecret: under point 8, I was a bit surprised. I could also enter my trial secret there. For an explicit community project, I'm a bit surprised.

Regards 

Michael

**Comment by binnes** [comment]
Hello Michael,
OKD is totally free and available for use by the community without having to register with RedHat.  However, OKD is built from the same source code as OpenShift Container Platform (ocp).  

The pull secret is required by the installer as ocp needs this to gain access to additional operator catalogs to enable enhanced features of ocp that are not available to OKD.   For OKD a 'fake' pull secret can be used or if you have your own operator catalog in a registry then you can provide the pull secret for that container registry (base64 encoded).   

The fake pull secret we provide in the [OKD documentation](https://www.okd.io/installation/#getting-started) is : `{"auths":{"fake":{"auth":"aWQ6cGFzcwo="}}}`

**Reply by lweller** [reply]
So I setup OKD with a pull secret form Red Hat. According to https://okd.io/docs/project/scos-migration-faq/ this seams to violate the Red Hat's Terms of Service. So what should I do now to be complient? Setup the Cluster from scratch with a fake pull secret?

**Reply by JaimeMagiera** [reply]
Yes, please.
